clippy_lints/src/loops.rs

```diff
@@ -2208,15 +2210,18 @@ impl<'a, 'tcx> Visitor<'tcx> for MutableVarsVisitor<'a, 'tcx> {
 }
 
 struct MutVarsDelegate {
-    mut_spans: HashMap<NodeId, Option<Span>>,
+    mut_spans: HashMap<NodeId, bool>,
 }
 
 impl<'tcx> MutVarsDelegate {
     fn update(&mut self, cat: &'tcx Categorization, sp: Span) {
-        if let Categorization::Local(id) = *cat {
-            if let Some(span) = self.mut_spans.get_mut(&id) {    
-                *span = Some(sp)
-            }
+        match *cat {
+            Categorization::Local(id) =>
+                if let Some(span) = self.mut_spans.get_mut(&id) {
+                    *span = true;
```

Comment:
Why can't we keep spans here? I think it would be great if we pointed to the variable instead of the entire loop

Reply:
Here the spans are for the variables that actually get mutated. So as soon as one is set, we do not lint. I figured a simple bool would better fit the usage .
(but I should indeed rename the field if we stick to a map from id to bool…)

Reply:
Sure! Updated :)

Reply:
Thanks!